github-helper.js

```diff
@@ -0,0 +1,66 @@
+exports.GithubHelper = class {
+  constructor({ client, context }) {
+    this.client = client;
+    this.context = context;
+  }
+
+  async assignToProjects({ columnName, newProjects }) {
+    const allProjects = await this.fetchProjects();
+    const columnIds = allProjects
+      .map((project) => project.node) // remove top-level "node" attribute
+      .filter(({ name: projectName }) => newProjects.includes(projectName))
+      .map(({ columns: { edges: columns } }) => (
+        columns.find((column) => column.node.name === columnName)
+          .node
+          .databaseId
+      ));
+    await this.addProjectCards({ columnIds });
+  }
+
+  async fetchProjects({ results, cursor } = { results: [] }) {
+    const query = `
+      query($cursor: String) {
+        repository(owner: "${this.context.repo.owner}", name: "${this.context.repo.repo}") {
+          projects(first: 100, after: $cursor) {
+            pageInfo {
+              hasNextPage
+              endCursor
+            }
+            edges {
+              node {
+                name
+                columns(first: 10) {
+                  edges {
+                    node {
+                      databaseId
+                      name
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    `;
+    const { repository: { projects } } = await this.client.graphql(query, { cursor });
+    results.push(...projects.edges);
+
+    if (projects.pageInfo.hasNextPage) {
+      await this.fetchProjects({ results, cursor: projects.pageInfo.endCursor });
+    }
+
+    return results;
```

Comment:
as you are waiting for recursive invocation here anyway, you don't need to have `results` as accumulator/memo argument.

```suggestion
    const results = [];
    results.push(...projects.edges);
    
    if (projects.pageInfo.hasNextPage) {
      const nextResults = await this.fetchProjects({ cursor: projects.pageInfo.endCursor })
      results.push(...nextResults);
    }

    return results;
```